base/compiler/abstractinterpretation.jl

```diff
@@ -259,6 +259,17 @@ function abstract_call_method_with_const_args(interp::AbstractInterpreter, @nosp
     inf_cache = get_inference_cache(interp)
     inf_result = cache_lookup(mi, argtypes, inf_cache)
     if inf_result === nothing
+        if edgecycle
+            # if there might be a cycle, check to make sure we don't end up
+            # calling ourselves here.
+            infstate = sv
+            while !(infstate === nothing)
+                if method === infstate.linfo.def
+                    return Any
+                end
+                infstate = infstate.parent
+            end
```

Comment:
```suggestion
            # if edgecycle is true, then there is a cycle and we're calling ourself
            return Any
```

Reply:
Well, that would just revert the original PR. What we need is to find out if adding the constant information breaks the cycle. Maybe we could do a full recursion check, but only look for other frames coming from constant prop?

Reply:
Okay, right we could probably make `edgecycle` computation more lenient to allow one identical parent to have the constant-prop flag set and still be good here.

Reply:
What is the best way to check that flag? `any(result.overridden_by_const)`?

Reply:
Sounds about right